### PR TITLE
global is not defined

### PR DESCRIPTION
**Pull request by benland**
global is not defined in the browser.
global && global.location throw an exception.

Angular CLI decided not to inject global to window.
https://github.com/unshiftio/url-parse/issues/119#issuecomment-413393181

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -53,7 +53,8 @@ var ignore = { hash: 1, query: 1 };
  * @public
  */
 function lolcation(loc) {
-  var location = global && global.location || {};
+  var globalVar = typeof window !== 'undefined' && typeof global === 'undefined' ? window : global;
```
Comment (lpinca) [inline index.js]:
```suggestion
  var globalVar;

  if (typeof window !== 'undefined')
    globalVar = window;
  else if (typeof global !== 'undefined')
    globalVar = global;
  else if (typeof self !== 'undefined')
    globalVar = self;
  else
    globalVar = {};
```

File: index.js [hunk 2]
```diff
@@ -53,7 +53,8 @@ var ignore = { hash: 1, query: 1 };
...
+  var location = globalVar && globalVar.location || {};
```
Comment (lpinca) [inline index.js]:
```suggestion
  var location = globalVar.location || {};
```

**Comment by 3rd-Eden** [comment]
YUP!

**Comment by 3rd-Eden** [comment]
Released as 1.4.4 -- Thanks for your contribution <3

**Pull request merged**